include/boost/geometry/strategies/geographic/closest_points.hpp

```diff
@@ -0,0 +1,286 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+
+// Copyright (c) 2019, Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_STRATEGIES_GEOGRAPHIC_CLOSEST_POINTS_HPP
+#define BOOST_GEOMETRY_STRATEGIES_GEOGRAPHIC_CLOSEST_POINTS_HPP
+
+#include <boost/geometry/algorithms/detail/closest_points/result.hpp>
+
+#include <boost/geometry/strategies/closest_points.hpp>
+//#include <boost/geometry/formulas/point_segment_distance.hpp>
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace closest_points
+{
+
+template
+<
+    typename FormulaPolicy = strategy::andoyer,
+    typename Spheroid = srs::spheroid<double>,
+    typename CalculationType = void
+>
+class geographic
+{
+public :
+
+    template <typename Point1, typename Point2>
+    struct calculation_type
+        : promote_floating_point
+          <
+              typename select_calculation_type
+                  <
+                      Point1,
+                      Point2,
+                      CalculationType
+                  >::type
+          >
+    {};
+
+    template <typename Point, typename PointOfSegment>
+    struct closest_point_result
+    {
+        typedef geometry::detail::closest_points::result
+                <
+                    typename calculation_type<Point, PointOfSegment>::type
+                > type;
+    };
+
+    inline geographic()
+        : m_spheroid()
+    {}
+
+    explicit inline geographic(Spheroid const& spheroid)
+        : m_spheroid(spheroid)
+    {}
+
+    Spheroid get_spheroid() const
```

Comment:
In other strategies there is member function `model()` for this. We could of course change it the other way around but we should be consistent. And then also the name wouldn't cover spheres returned by spherical strategies.

Reply:
sure, renamed to model